### PR TITLE
Refactor Flump View construction

### PR DESCRIPTION
**Pull request by carlos-alberto**
After working for some time with flump, and building out more
complex api's we've noticed limitations in the way we
construct our views.

With the current setup we favour a lot of inheritance, which is
fine for simpler api's, however with more complex api's it becomes
more difficult to check which methods are implemented for which views,
and how the orm logic is applied.

This commit refactors the methods to be defined as a set on the view,
rather than inheriting the methods you wish to use. This should make
it easier to identify exactly which methods are defined for a view.

We also move the logic for getting entities and their counts to a new
`Fetcher` class, which should encapsulate all of the logic needed to
get entitites.

We move the logic for deleting/updating/creating entities to a new
`OrmIntegration` class, again encapsulating this logic in one place,
rather than over different classes.

The major point of this refactor is to take more of a "composition"
approach rather than an "inheritance" approach, hopefully simplifying
the logic required to build a Flump app in the process.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -122,6 +101,57 @@ The View should also define ``RESOURCE_NAME`` and ``SCHEMA`` properties.
...
+The Fetcher
+===========
+
+To get data from the database we must define a class which inherits from
+``flump.Fetcher`` and provides the following methods:
+
+
+* ``get_entity``, which retrieves a singular entity given an ``entity_id``.
+
+* ``get_many_entities``, which returns all of the entities available. If you would like to paginate the entities, we provide a mixin for this purpose. See :ref:`pagesizepagination`.
```
Comment (obmarg) [inline README.rst]: Is this pagesizepagination bit still accurate?

File: README.rst [hunk 2]
```diff
@@ -122,6 +101,57 @@ The View should also define ``RESOURCE_NAME`` and ``SCHEMA`` properties.
...
+The View
+=========
+
+We can then tie these together to define our view. Our view must inherit
+from ``flump.FlumpView``, and define the following properties:
+
+* ``FETCHER``, the class we use to get entities.
+* ``ORM_INTEGRATION``, the class we use to update/create/delete entities.
+* ``SCHEMA``, schema which we use to marhsal/unmarshal the data.
+* ``RESOURCE_NAME``, the name of the resource, used to define the URL.
```
Comment (obmarg) [inline README.rst]: How is this used to define the URL?  And could do with explaining why we _also_ need to specify the URL...

File: flump/__init__.py [hunk 3]
```diff
@@ -69,16 +74,35 @@ def register_flump_view(self, view_class, url):
...
-                          methods=('GET', 'PATCH', 'DELETE'))
+        get_many_func = view_func if 'GET_MANY' in methods else _meth_not_allowed  # noqa
+        self.add_url_rule(url, methods=('GET',),
+                          view_func=get_many_func, strict_slashes=False)
+
+        post_func = view_func if 'POST' in methods else _meth_not_allowed
+        self.add_url_rule(url, methods=('POST',),
+                          view_func=post_func, strict_slashes=False)
+
+        all_entity_methods = {'GET', 'PATCH', 'DELETE'}
```
Comment (obmarg) [inline flump/__init__.py]: Should we be using the constants here?

**Comment by lukasz-madon** [comment]
There are *.DS_Store files in the PR

**Comment by obmarg** [comment]
Just a thought: is there a way we could stop explicitly passing around the `entity_id` parameter?  Occasionally it's useful to have an endpoint that just represents a single resource, so `entity_id` doesn't _always_ make sense.  It's always available via flasks `request.view_args`, so seems like it might be possible?

**Pull request merged**